client/packages/android/app/src/main/java/org/openmsupply/client/NativeApi.java

```diff
@@ -64,12 +78,120 @@ public String getLocalUrl() {
+                        // actually no point - the timeout only applies when trying to find a server
+                        // when using localhost it returns immediately even if the server isn't
+                        // responding
+                        urlc.setConnectTimeout(1000);
+                        urlc.connect();
+                        if (urlc.getResponseCode() == 200) {
+                            isServerRunning = true;
+                        }
+                    } catch (SSLHandshakeException e) {
+                        // server is running and responding with an SSL error
+                        // which we will ignore, so ok to proceed
+                        isServerRunning = true;
+                    } catch (Exception e) {
+                        Log.e("omSupply", e.getMessage());
+                        isServerRunning = false;
+                    }
+                    retryCount--;
+                    sleep(1000);
+                }
+
+                // .post to run on UI thread in the two calls below
+                if (isServerRunning) {
+                    webView.post(() -> webView.loadUrl(localUrl + "/android"));
+                } else {
+                    webView.post(() -> webView.loadData(ErrorPage.encodedHtml, "text/html", "base64"));
+                    hideSplash();
+                }
+            }
+        });
+        thread.start();
+    }
+
+    private Drawable getSplashDrawable(Context context) {
+        int splashId = context.getResources().getIdentifier("splash", "drawable", context.getPackageName());
+        try {
+            Drawable drawable = context.getResources().getDrawable(splashId, context.getTheme());
+            return drawable;
+        } catch (Resources.NotFoundException ex) {
+            Log.w("omSupply", "No splash image found");
+            return null;
+        }
+    }
+
+    private void showSplashScreen() {
+        if (this.splashScreen != null) {
+            return;
+        }
+
+        Bridge bridge = this.getBridge();
+        Context context = bridge.getContext();
+        Activity activity = bridge.getActivity();
+
+        activity.runOnUiThread(
+                () -> {
+                    this.splashScreen = new Dialog(activity, R.style.capacitor_immersive_style);
+
+                    Drawable splash = getSplashDrawable(context);
+                    LinearLayout parent = new LinearLayout(context);
+                    parent.setLayoutParams(
+                            new LinearLayout.LayoutParams(LinearLayout.LayoutParams.MATCH_PARENT,
+                                    LinearLayout.LayoutParams.WRAP_CONTENT));
+                    parent.setOrientation(LinearLayout.VERTICAL);
+                    if (splash != null) {
+                        parent.setBackground(splash);
+                    }
+                    this.splashScreen.setContentView(parent);
```

Comment:
can simplify this slightly with:

https://github.com/openmsupply/open-msupply/compare/1190-example1...1190-example2

Although I kind of like the encoding better, and it also solves the problem with flicker i described (I think)